lighter/system.py

```
@@ -175,6 +175,10 @@ def _calculate_metrics(self, input: Any, target: Any, pred: Any) -> Any | None:
             metrics = getattr(self.metrics, self.mode)
             if metrics is not None:
                 adapters = getattr(self.adapters, self.mode)
+                try:
+                    metrics = metrics.to(input.device)
+                except AttributeError:
+                    pass
```

Ruff (0.8.2)

Use `contextlib.suppress(AttributeError)` instead of `try`-`except`-`pass`

Replace with `contextlib.suppress(AttributeError)`

(SIM105)